c++/src/kj/CMakeLists.txt

```
@@ -163,6 +163,29 @@ if(NOT CAPNP_LITE)
   install(FILES ${kj-http_headers} DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/kj/compat")
 endif()
 
+# kj-tls ======================================================================
+set(kj-tls_sources
+  compat/readiness-io.c++
+  compat/tls.c++
+)
+set(kj-tls_headers
+  compat/readiness-io.h
+  compat/tls.h
+)
+if(NOT CAPNP_LITE)
+  add_library(kj-tls ${kj-tls_sources})
+  add_library(CapnProto::kj-tls ALIAS kj-tls)
+  target_link_libraries(kj-tls PUBLIC kj-async kj)
+  if (WITH_OPENSSL)
+    target_compile_definitions(kj-tls PRIVATE KJ_HAS_OPENSSL)
+    target_link_libraries(kj-tls PUBLIC OpenSSL::SSL OpenSSL::Crypto)
```

```suggestion
    target_link_libraries(kj-tls PRIVATE OpenSSL::SSL OpenSSL::Crypto)
```
`OpenSSL` headers are not included in `kj-tls` public headers